src/python/pants/backend/codegen/protobuf/python/rules_integration_test.py

```diff
@@ -106,7 +112,8 @@ def test_generates_python(self) -> None:
             ),
         )
         self.add_to_build_file(
-            "src/protobuf/dir2", "protobuf_library(dependencies=['src/protobuf/dir1'])"
+            "src/protobuf/dir2",
+            "protobuf_library(dependencies=['src/protobuf/dir1'], python_source_root='src/python')",
```

Comment:
It would be good to still test the default behavior as well.

Reply:
It can be a really simple test like `test_top_level_source_root` is.

Reply:
This does test the default behavior, via `src/protobuf/dir1`.  Or am I missing your point?

Reply:
Not quite. I mean adding a test where you leave off `python_source_root`. I want to make sure that that still behaves like we'd expect, given that it's a different branch in the `if/else` block.

Reply:
Right, there is no `python_source_root` in the `protobuf_library` at `src/protobuf/dir1` (https://github.com/benjyw/pants/blob/a7c33b428f4c383a74dfc2cbe99aa91212934c70/src/python/pants/backend/codegen/protobuf/python/rules_integration_test.py#L100) and we verify that the files are generated under `src/protobuf/` .

Reply:
@Eric-Arellano is this sufficient? Or am I still missing something.

Reply:
Ah I see now. Sorry I missed that. And thank you for adding the new test for a top-level root!